### ISSUE TITLE
Implement NetTcpBinding Security.Transport negative E2E scenarios invalid cert

### ISSUE THREAD

**Issue by hongdai** [dotnet/wcf#370]
This issue tracks implementation of the following test
 NetTcpBinding Security.Transport negative E2E scenarios
b) ii, iii, iv, vi

in test plan #322 .

**Comment by zhenlan** [comment]
This is currently blocked by issue #308 which @iamjasonp is working on.

**Comment by iamjasonp** [comment]
What is the definition of "invalid cert" here? Is it a cert that is: 
* Valid but cert is not made for the machine? 
* Expired? 
* Revoked? 
* Not yet valid? 
* Valid but root isn't trusted? 
* Weak algorithm (e.g., md5 vs. sha1, < 2048-bit key)? 

These considerations will need to be taken into account when figuring out how to generate our certs and how we might go about testing them.

**Comment by hongdai** [comment]
Please refer to NetTcpBinding Security.Transport negative E2E scenarios
 b) ii, iii, iv, vi

in test plan #322 for the detailed invalid cases we plan to cover, pending the availability of generating them per Jason's comment above.

**Comment by roncain** [comment]
Which of the invalid cert cases are handled by WCF code?  I think any negative cert scenario that triggers unique WCF code should be tested at some point.  But negative cases handled at some other layer (including either the crypto layers or by the Windows WCF server-side code) probably count as "nice to have".  Do we know what negative scenarios are being tested by the crypto tests?

**Comment by hongdai** [comment]
Agreed. I will take a look at the WCF code and prioritize those negative cert scenarios triggers unique WCF code higher.
